File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        heap = [] # (Len to 0, point)
        out = []

        for point in points:
            heapq.heappush(heap, (math.sqrt(pow(point[0], 2) + pow(point[1], 2)), point))
        for i in range(k):
            elem = heapq.heappop(heap)
            out.append([elem[1][0], elem[1][1]])

        return out

File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.cs
public class Solution {
    public int[][] KClosest(int[][] points, int k)
    {
        var pq = new PriorityQueue<int[], double>();

        foreach (var p in points)
        {
            double x = p[0];
            double y = p[1];
            double dist = Math.Sqrt(Math.Pow(x, 2) + Math.Pow(y, 2));
            pq.Enqueue(p, dist); // minheap
        }

        int[][] result = new int[k][];
        for (int i = 0; i < k; i++)
        {
            result[i] = pq.Dequeue();
        }

        return result;
    }
}
